test/erc677/ERC677.test.ts

```diff
@@ -0,0 +1,110 @@
+import {deployments, ethers, getNamedAccounts} from 'hardhat';
+import {BigNumber} from '@ethersproject/bignumber';
+import {expect} from '../chai-setup';
+import {Contract} from 'ethers';
+
+describe('ERC677Token', function () {
+  let tokenReceiver: Contract;
+  let gemToken: Contract;
+  let emptyContract: Contract;
+  let fallbackContract: Contract;
+
+  async function createFixture() {
```

Comment:
This should use `deployments.createFixture` to speed up tests see : https://github.com/thesandboxgame/sandbox-private-contracts/blob/b14f22fcc2e270db7ab211cebf9b05da8c8d9d80/test/permit/fixtures.ts#L9